### PR TITLE
feat(ecocredit): add batch denom migration

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #1049 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/migrations/v3/keys.go [hunk 1]
```diff
@@ -92,11 +94,20 @@ func IterateSupplies(store sdk.KVStore, storeKey byte, cb func(denom, supply str
...
 
-// Calculate the ID to use for a new project, based on the class id and
-// the project sequence number.
-//
-// The initial version has format:
-// <class id><project seq no>
-func FormatProjectID(classID string, projectSeqNo uint64) string {
-	return fmt.Sprintf("%s%02d", classID, projectSeqNo)
+func parseBatchDenom(denom string) (*time.Time, *time.Time, error) {
+	// batch denom format: <class id>-<start date>-<end date>-<batch seq no>
```
Comment (technicallyty) [inline x/ecocredit/migrations/v3/keys.go:98]:
```suggestion
	// batch denom format: <project id>-<start date>-<end date>-<batch seq no>
```

File: x/ecocredit/migrations/v3/core_test.go [hunk 2]
```diff
@@ -292,3 +318,21 @@ func TestMigrations(t *testing.T) {
 	require.Nil(t, bz)
 
 }
+
+func formatBatchDenom(classId string, batchSeqNo uint64, startDate *time.Time, endDate *time.Time) string {
+	return fmt.Sprintf(
+		"%s-%s-%s-%03d",
+
+		// Class ID string
+		classId,
```
Comment (technicallyty) [inline x/ecocredit/migrations/v3/core_test.go:327]:
```suggestion
func formatBatchDenom(projectId string, batchSeqNo uint64, startDate *time.Time, endDate *time.Time) string {
	return fmt.Sprintf(
		"%s-%s-%s-%03d",

		// Project ID string
		projectId,
```

**Review by technicallyty — APPROVED** [review]
pre-approving, just 2 small nits 👍🏻

File: x/ecocredit/migrations/v3/keys.go (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/migrations/v3/keys.go:98]: This is the version of the batch denom we are migrating from.

File: x/ecocredit/migrations/v3/core_test.go (hunk 2 above)
Comment (ryanchristo) [reply x/ecocredit/migrations/v3/core_test.go:327]: Same here. This is the version we are migrating from.

**Pull request merged**